leaderelection/context.go

```diff
@@ -173,8 +173,10 @@ func newStandardBuckets(queueName string, cc ComponentConfig) []reconciler.Bucke
 	bs := hash.NewBucketSet(names)
 
 	bkts := make([]reconciler.Bucket, 0, cc.Buckets)
-	for name := range names {
-		bkts = append(bkts, hash.NewBucket(name, bs))
+	for _, name := range names.List() {
+		// The error shouldn't happen as we make sure that `name` is in the BucketSet.
+		bkt, _ := bs.NewBucket(name)
+		bkts = append(bkts, bkt)
 	}
 	return bkts
```

Comment:
If there's an error path, I'd rather we checked the error.  My point was that if we had a method like `BucketList` that returned `[]Bucket` instead of `[]string` then this could simply be:

```
return bs.ActualBucketList()
```

Reply:
This can naturally avoid the error path because it knows the bucket `.Has()` the name.  To ignore the error is to know the details behind the abstraction, which limits our ability to change that implementation in a sane way 😅

Reply:
Introduced `ActualBucketList`. PTAL